types/astraled/airQualityLuminary_V0_02/uplink.js

```diff
@@ -0,0 +1,224 @@
+  var e = bits_float >>> 23 & 0xff;
+  var m = (e === 0) ? (bits_float & 0x7fffff) << 1 : (bits_float & 0x7fffff) | 0x800000;
+  var f = sign * m * Math.pow(2, e - 150);
+  return f;
+}
+
+function get_byte(buff, nr) {
+  return Bits.bitsToUnsigned(buff.substr(nr * 8, 8));
+}
+
+function get_word(buff, nr) {
+  return swap16(Bits.bitsToUnsigned(buff.substr(nr * 8, 16)));
+}
+
+function get_long(buff, nr) {
+  return swap32(Bits.bitsToUnsigned(buff.substr(nr * 8, 32)));
+}
+
+function get_float(buff, nr) {
+  return bytesToFloat(get_long(buff, nr));
+}
+
+function consume(event) {
+
+  var payload_hex = event.data.payload_hex;
+  var buff = Bits.hexToBits(payload_hex);
+
+  var data_size = payload_hex.length / 2;
+
+  var data = {};
+  var rd_pos = 0;
+
+  data.rssi = event.data.flags.AK_RSSI;
+
+  do {
+    var para_size = get_byte(buff, rd_pos++) & 0x3F;  // get size
+    var para_type = get_byte(buff, rd_pos++);  // get type
+
+    switch (para_type) {
+      case 1:
+        data.protocol_version = "Version: " + get_byte(buff, rd_pos + 1) + "." + get_byte(buff, rd_pos);
+        break;
+      case 2:
+        data.app_version = "Version: " + get_byte(buff, rd_pos + 2) + "." + get_byte(buff, rd_pos + 1) + "p" + get_byte(buff, rd_pos);
+        break;
+      case 3:
+        data.lora_stack_version = "Version: " + get_byte(buff, rd_pos + 2) + "." + get_byte(buff, rd_pos + 1) + "." + get_byte(buff, rd_pos);
+        break;
+      case 4:
+        data.lora_version = "Version: " + get_byte(buff, rd_pos + 2) + "." + get_byte(buff, rd_pos + 1) + "." + get_byte(buff, rd_pos);
+        break;
+      case 6:
+        switch (get_byte(buff, rd_pos)) {
+          case 1: data.msg_cycle_time_1 = get_word(buff, rd_pos + 1); break;
+          case 2: data.msg_cycle_time_2 = get_word(buff, rd_pos + 1); break;
+          case 3: data.msg_cycle_time_3 = get_word(buff, rd_pos + 1); break;
+          case 4: data.msg_cycle_time_4 = get_word(buff, rd_pos + 1); break;
+          case 5: data.msg_cycle_time_5 = get_word(buff, rd_pos + 1); break;
+          case 6: data.msg_cycle_time_6 = get_word(buff, rd_pos + 1); break;
+          case 7: data.msg_cycle_time_7 = get_word(buff, rd_pos + 1); break;
+          case 8: data.msg_cycle_time_8 = get_word(buff, rd_pos + 1); break;
+        }
+        break;
+      case 7:
+        data.group_address = get_byte(buff, rd_pos);
+        break;
+      case 8:
+        data.serial_number = get_long(buff, rd_pos);
+        break;
+      case 9:
+        data.temperature = get_word(buff, rd_pos) / 100;
+        break;
+      case 10:
+        data.humidity = get_word(buff, rd_pos) / 10;
+        break;
+      case 11:
+        data.voc = get_word(buff, rd_pos);
+        break;
+      case 12:
+        data.co2 = get_word(buff, rd_pos);
+        break;
+      case 13:
+        data.eco2 = get_word(buff, rd_pos);
+        break;
+      case 14:
+        data.iaq_state_int = get_byte(buff, rd_pos);
```

Comment:
add missing props to the schema

Reply:
Do we really need everything in the schema? thats gonna be xtra long xd

Reply:
yes it should include the full schema